From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Mon, 14 Jun 2021 15:53:28 -0700
Subject: [PATCH] Restore previous cmake checks for GL/EGL

The new cmake changes to use FindOpenGL exclusively do not appear to work on
systems (like Clear Linux OS) that lack GLVND. Restore the previous behavior to
fix the checks.

[2021-09-03]: The EGL detection is working now in the absence of GLVND,
              but GL/GLX detection still fails.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 CMakeLists.txt                             | 5 ++---
 src/declarativeimports/core/CMakeLists.txt | 2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7787a05..ac5e4db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,9 +110,8 @@ set_package_properties(EGL PROPERTIES
                        TYPE OPTIONAL
                       )
 
-add_feature_info(GLX ${OpenGL_GLX_FOUND} "OpenGL GLX libraries.")
-if(OpenGL_GLX_FOUND AND X11_FOUND AND (Qt5Gui_OPENGL_IMPLEMENTATION STREQUAL "GL"))
-    set(HAVE_GLX 1)
+if(OPENGL_FOUND AND X11_FOUND AND (Qt5Gui_OPENGL_IMPLEMENTATION STREQUAL "GL"))
+    set(HAVE_GLX ${HAVE_X11})
 else()
     set(HAVE_GLX 0)
 endif()
diff --git a/src/declarativeimports/core/CMakeLists.txt b/src/declarativeimports/core/CMakeLists.txt
index c95b4b4..066f8cb 100644
--- a/src/declarativeimports/core/CMakeLists.txt
+++ b/src/declarativeimports/core/CMakeLists.txt
@@ -54,7 +54,7 @@ if(HAVE_X11)
   endif()
 
   if(HAVE_GLX)
-    target_link_libraries(corebindingsplugin OpenGL::GLX)
+    target_link_libraries(corebindingsplugin ${OPENGL_gl_LIBRARY})
   endif()
 endif()
 
